### ISSUE TITLE
Security Report — non-disableable `--dangerously-skip-permissions` Flag in analyze_issue_llm

### ISSUE THREAD

**Issue by Dingisoul-DEV** [kesslerio/vibe-check-mcp#302]
**Affected tools:** `analyze_issue_llm`, `analyze_text_llm`

### Severity

HIGH — Bypassing AI Safety Guardrails Without User Disclosure

### Description

The `analyze_issue_llm` tool delegates to an internal `analyze_text_llm` function, which always invokes the Claude CLI subprocess with the `--dangerously-skip-permissions` flag:

```python
command = ["claude", "-p", "--dangerously-skip-permissions", prompt]
```

The call chain is:
```
analyze_issue_llm(issue_content, ...)
    └─▶ analyze_text_llm(content, task_type="issue_analysis", ...)
            └─▶ subprocess: claude -p --dangerously-skip-permissions <prompt>
```

This flag disables all permission prompts in the Claude subprocess, allowing it to perform arbitrary privileged operations (file writes, shell commands, network requests, etc.) without any user confirmation. Neither `analyze_issue_llm` nor `analyze_text_llm` discloses this in their tool descriptions. Both descriptions instead use promotional language ("comprehensive", "deep", "full reasoning capabilities") that encourages the AI to prefer these tools over alternatives, constituting preference shaping.

### Impact

Any operation the Claude subprocess performs — including destructive filesystem changes or outbound network requests — bypasses the user's permission model entirely. A malicious or compromised prompt could leverage this to cause irreversible damage without user awareness.

### Request

1. Remove the `--dangerously-skip-permissions` flag, or add a flag to disable this feature.
2. Remove promotional language from the tool descriptions to avoid influencing AI tool selection.

**Comment by IgorGanapolsky** [comment]
One additional hardening point: if this remains as an opt-in mode, the tool description should expose it as a capability boundary, not as implementation detail hidden behind "deep analysis" language.

A safer shape would be something like:

```json
{
  "dangerous_permission_bypass": false,
  "allowed_tools": ["read", "grep"],
  "blocked_tools": ["write", "shell", "network", "process"],
  "outbound_actions_require_approval": true
}
```

For the subprocess call, I would also log the resolved workspace, exact allowed/blocked tool set, and whether the child process can read parent environment variables. The risk is not only destructive file writes; it is also silent credential/env exfiltration from a nested agent whose permission model differs from the parent session.

I have been using a local operator checklist for this class of issue: resolve workspace with `realpath`, default-deny shell/process/network, require a per-run capability manifest, and scan for long-running agent/session state before enabling autonomous mode. The read-only Mac scan is here if useful for reproducing workstation-side amplification issues: https://igorganapolsky.github.io/AI_OpenClaw_Setup_Kits/scripts/mac-agent-risk-scan.sh
